ftfy/formatting.py

```diff
@@ -0,0 +1,141 @@
+from unicodedata import east_asian_width, combining, category, normalize
+
+
+def character_width(char):
+    """
+    Determine the width that a character is likely to be displayed as in
+    a monospaced terminal. The width will always be 0, 1, or 2.
+
+    We are assuming that the character will appear in a modern, Unicode-aware
+    terminal emulator, where CJK characters -- plus characters designated
+    "fullwidth" by CJK encodings -- will span two character cells. (Maybe
+    it's an over-simplification to call these displays "monospaced".)
+
+    When a character width is "Ambiguous", we assume it to span one character
+    cell, because we assume that the monospaced display you're using is
+    designed mostly to display Western characters with CJK characters as the
+    exception. This assumption will go wrong, for example, if your display is
+    actually on a Japanese flip-phone... but let's assume it isn't.
+
+    If this character is a control code, the idea of it having a "width"
```

Comment:
This doesn't seem to really capture what the function is doing, right?  It's more like some of the control codes are known to have width zero, and some are punted in the way being described.